configuration/secrets.rst

```diff
@@ -0,0 +1,320 @@
+
+This will generate a pair of asymmetric **cryptographic keys**. Each
+:ref:`environment <configuration-environments>` has its own set of keys. Assuming
+you're coding locally in the ``dev`` environment, this will create:
+
+``config/secrets/dev/dev.encrypt.public.php``
+    Used to encrypt/add secrets to the vault. Can be safely committed.
+
+``config/secrets/dev/dev.decrypt.private.php``
+    Used to decrypt/read secrets from the vault. The ``dev`` decryption key can
+    be committed (assuming no highly-sensitive secrets are stored in the dev vault)
+    but the ``prod`` decryption key should *never* be committed.
+
+You can generate a pair of cryptographic keys for the ``prod`` environment by
+running:
+
+.. code-block:: terminal
+
+    $ php bin/console secrets:generate-keys --env=prod
+
+This will generate ``config/secrets/prod/prod.encrypt.public.php`` and
+``config/secrets/prod/prod.decrypt.private.php``.
+
+.. caution::
+
+    The ``prod.decrypt.private.php`` file is highly sensitive. Your team of developers
+    and even Continuous Integration services don't need that key. If the
+    **decryption key** has been exposed (ex-employee leaving for instance), you
+    should consider generating a new one by running:
+    ``secrets:generate-keys --rotate``.
+
+.. _secrets-set:
+
+Create or Update Secrets
+------------------------
+
+Suppose you want to store your database password a secret. By using the
+``secrets:set`` command, you should add this secret to both the ``dev`` *and*
+``prod`` vaults:
+
+.. code-block:: terminal
+
+    # the input is hidden as you type for security
+
+    # set your a default development value (can be overridden locally)
+    $ php bin/console secrets:set DATABASE_PASSWORD
+
+    # set your production value
+    $ php bin/console secrets:set DATABASE_PASSWORD --env=prod
+
+This will create a new file for the secret in ``config/secrets/dev`` and another
+in ``config/secrets/prod``. You can also set the secret in a few other ways:
+
+.. code-block:: terminal
+
+    # provide a file where to read the secret from
+    $ php bin/console secrets:set DATABASE_PASSWORD ~/Download/password.json
+
+    # or contents passed to STDIN
+    $ echo -n "$DB_PASS" | php bin/console secrets:set DATABASE_PASSWORD -
+
+    # or let Symfony generate a random value for you
+    $ php bin/console secrets:set REMEMBER_ME --random
+
+Referencing Secrets in Configuration Files
+------------------------------------------
+
+Secret values can be referenced in the same way as
+:ref:`environment variables<config-env-vars>`. Be careful that you don't
+accidentally define a secret *and* an environment variable with the same name:
+**environment variables override secrets**.
+
+If you stored a ``DATABASE_PASSWORD`` secret, you can reference by:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/doctrine.yaml
+        doctrine:
+            dbal:
+                password: '%env(DATABASE_PASSWORD)%'
```

Comment:
Shouldn't this be `%env(secret:DATABASE_PASSWORD)` ?

Reply:
there is no `secret` env processor, the document is right here, to access secret, you should use `%env(DATABASE_PASSWORD)%`. this is explain in the paragraph above (line 104).